llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -4546,14 +4548,18 @@ ParseStatus AArch64AsmParser::tryParseZTOperand(OperandVector &Operands) {
     if (getParser().parseExpression(ImmVal))
       return ParseStatus::NoMatch;
     const MCConstantExpr *MCE = dyn_cast<MCConstantExpr>(ImmVal);
+    Operands.push_back(AArch64Operand::CreateImm(
+        MCConstantExpr::create(MCE->getValue(), getContext()), StartLoc,
+        getLoc(), getContext()));
     if (!MCE)
       return TokError("immediate value expected for vector index");
+    if (getTok().is(AsmToken::Comma)) {
```

Comment:
Can you write instead as:
  if (parseOptionalToken(AsmToken::Comma)
    if (parseOptionalMulOperand(Operands))
        return MatchOperand_ParseFail;